packages/frontend-plugin-api/src/analytics/useAnalytics.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useApi } from '@backstage/core-plugin-api';
+import { useAnalyticsContext } from './AnalyticsContext';
+import { analyticsApiRef, AnalyticsTracker, AnalyticsApi } from '../apis';
+import { useRef } from 'react';
+import { Tracker } from './Tracker';
+
+function useAnalyticsApi(): AnalyticsApi {
+  try {
+    return useApi(analyticsApiRef);
+  } catch {
+    return { captureEvent: () => {} };
+  }
+}
+
+/**
+ * Gets a pre-configured analytics tracker.
+ *
+ * @public
+ */
+export function useAnalytics(): AnalyticsTracker {
```

Comment:
As discussed separately, let's aim to have as many things as possible imported directly from `core-plugin-api` with a relative import in order to avoid code duplication. Similar to #20992. I'll try to summarize the strategy and reasoning here over in #19545

Reply:
Coming back to this I think there are going to be enough differences here that it makes sense to duplicate and refactor straight away.

Reply:
Updated: https://github.com/backstage/backstage/pull/21244/commits/5c794cd5835c6c2be565134d0a744743dc95637a